docs/plugins/overview.mdx

```diff
@@ -1,33 +1,41 @@
 ---
-title: Engine Plugins
+title: Basics
 sidebar_position: 1
 description: "Learn about Hasura's powerful plugins architecture."
-sidebar_label: Overview
 keywords:
   - hasura plugins
   - allowlist plugin
   - plugins architecture
   - engine plugins
-hide_table_of_contents: true
 seoFrontMatterUpdated: false
 ---
 
-import { OverviewPlainCard } from "../../src/components/OverviewPlainCard";
-import { OverviewTopSectionIconNoVideo } from "@site/src/components/OverviewTopSectionIconNoVideo";
-import Icon from "@site/static/icons/remote-schema.svg";
-
 # Engine Plugins
 
 ## Introduction
 
-Hasura gives you the power to add HTTP hooks to different execution steps of your supergraph. This allows you to extend
-the capabilities of your API by adding custom logic executed in response to events.
+Hasura's engine plugins architecture allows you to integrate HTTP hooks at various stages of the API execution pipeline.
+This functionality enables you to embed custom logic that reacts to specific events, effectively extending the core
+capabilities of your API. These hooks can be triggered either before or after the execution of a query or mutation,
+providing a powerful mechanism to implement user-defined workflows and event-driven customizations.
+
+## Types of plugins
+
+### Pre-Parse plugins
+
+Pre-parse plugins are executed at the very beginning of the execution pipeline. They allow you to insert custom logic
+**before** the query-parsing step occurs. This enables you to pre-process, or manipulate incoming queries, and modify
+the behavior of your API at an early stage.
+
+### Pre-Response plugins
 
-Engine plugins allow you to execute this logic either before or after the execution of a query or a mutation. This gives
-you the power to add custom logic driven by user-defined events on your API.
+Pre-response plugins are triggered at the final stage of the execution pipeline, **after the query has been executed but
+before the response is sent to the client**. These plugins allow you to apply post-processing logic to the results, such
+as formatting, enriching data, implementing additional transformations, or calling third-party services before the
+client receives the response.
```

Comment:
The pre-response plugins cannot modify the response sent back to the client. They can only get the notification from the DDN about the response (request and other info), that we are sending to the client.